src/services/crawl-dids/dids.service.ts

```diff
@@ -38,25 +30,44 @@ export default class DidDatabaseService extends BullableService {
             throw new Error('DID is required for deletion');
         }
 
-        this.logger.info(`Attempting to delete DID: ${ctx.params}`);
+        this.logger.info(`Attempting to delete DID: ${did}`);
 
-        const deletedRows = await knex('dids').where({ did }).del();
+        const deletedRows = await knex('dids')
+            .where({ did })
+            .update({ is_deleted: true, deleted_at: new Date().toISOString(), event_type: "remove_did" });
 
         if (deletedRows === 0) {
             this.logger.warn(`No DID found to delete: ${did}`);
             return { success: false, message: `No record found for DID: ${did}` };
         }
 
-        this.logger.info(`Deleted DID: ${did}`);
+        this.logger.info(`Marked DID as deleted: ${did}`);
         return { success: true };
     }
 
-
-    @Action({
-        name: 'get'
-    })
+    @Action({ name: 'get' })
     async getDid(ctx: any) {
         const { did } = ctx.params;
         return await knex('dids').where({ did }).first();
     }
-}
\ No newline at end of file
+
+
+    @Action({ rest: "GET get/:did", params: { did: "string" } })
+    async getSingleDid(ctx: Context<{ did: string }>) {
+        const did = await knex("dids").where({ did: ctx.params.did }).first();
+        if (!did) throw new Error("DID not found");
+        return did;
+    }
+
+    @Action({ rest: "GET list", params: { page: { type: "number", optional: true, default: 1 }, limit: { type: "number", optional: true, default: 20 } } })
```

Comment:
The spec defines some parameters that are not considered here: https://verana-labs.github.io/verifiable-trust-vpr-spec/#mod-dd-qry-1-list-dids

It does not mention anything about pagination, but probably we should consider it. It does include a response_max_size but I'm not sure if it is used for pagination.